### PR TITLE
chore(rust): report the TableNotFound error while dropping non-exist table

### PR DESCRIPTION
**Pull request by BubbleCal**
this will work after upgrading lance with https://github.com/lancedb/lance/pull/1995 merged
see #884 for details

### CODE REVIEW

**Comment by BubbleCal** [comment]
I will enable the tests after upgrading lance, for now it's blocked by some changes

File: rust/lancedb/src/connection.rs [hunk 1]
```diff
@@ -691,13 +699,49 @@ mod tests {
...
+    #[tokio::test]
+    #[ignore = "this can't pass due to https://github.com/lancedb/lancedb/issues/1019, enable it after the bug fixed"]
+    async fn test_open_table() {
+        let tmp_dir = tempdir().unwrap();
+        let uri = tmp_dir.path().to_str().unwrap();
+        let db = connect(uri).execute().await.unwrap();
+
+        assert_eq!(db.table_names().await.unwrap().len(), 0);
+
+        db.open_table("invalid_table").execute().await.unwrap();
```
Comment (westonpace) [inline rust/lancedb/src/connection.rs:711]: Won't this `unwrap` cause a panic since the table doesn't exist?  Should this line be removed?

**Review by westonpace — APPROVED** [review]
We should wait for the lance fixes to merge and the tests to pass before merging but the fix looks good.

File: rust/lancedb/src/connection.rs (hunk 1 above)
Comment (BubbleCal) [reply rust/lancedb/src/connection.rs:711]: you are right, will fix this in the enabling PR

**Pull request merged**